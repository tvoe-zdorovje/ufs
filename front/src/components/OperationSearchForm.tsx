import React from 'react'
import {Col, Form, Input, Row, Select} from 'antd';
const FormItem = Form.Item;
const Option = Select.Option;

const styles = require('./Shared.less');

export interface OperationSearchFormProps {
}

export default class OperationSearchForm extends React.Component<OperationSearchFormProps, any> {

  render() {

    const input = (placeholder?) => (
      <FormItem>
        <Input placeholder={placeholder}/>
      </FormItem>
    );

    const label = (value) => (<span className={styles.label}>{value}</span>);

    return (
      <div>
        <Row>
          {label('Введите данные по клиенту')}
        </Row>
        <Row style={{marginTop: 15}}>
          <Col span={4}>
            {label('Фамилия')}
          </Col>
          <Col offset={1} span={16}>
            {input()}
          </Col>
        </Row>
        <Row>
          <Col span={4}>
            {label('Имя')}
          </Col>
          <Col offset={1} span={16}>
            {input()}
          </Col>
        </Row>
        <Row>
          <Col span={4}>
            {label('Отчество')}
          </Col>
          <Col offset={1} span={16}>
            {input()}
          </Col>
        </Row>
        <Row style={{marginTop: 15}}>
          {label('Введите данные ДУЛ: ')}
        </Row>
        <Row>
          <Col span={4}>
            <span className={styles.label}>Тип ДУЛ</span>
          </Col>
          <Col offset={1} span={16} >
            <Select defaultValue="1" style={{display: 'block'}}>
              <Option value="1">Паспорт гражданина РФ</Option>
              <Option value="2">Загранпаспорт гражданина РФ</Option>
              <Option value="3">Военный билет</Option>
              <Option value="4">Паспорт моряка</Option>
            </Select>
          </Col>
        </Row>
        <Row>
          <Col className={styles.label} span={4}>
            {input('Серия')}
          </Col>
          <Col offset={1} span={16}>
            {input('Номер')}
          </Col>
        </Row>
      </div>
    );
  }
}
