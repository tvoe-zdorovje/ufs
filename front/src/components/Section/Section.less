@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
