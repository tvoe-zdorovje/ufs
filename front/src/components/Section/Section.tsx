import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
