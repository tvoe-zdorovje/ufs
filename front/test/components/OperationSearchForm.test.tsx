import React from 'react'
import {
  shallow,
  render
} from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  Table,
  Button,
  AutoComplete,
  Form,
} from 'antd'

const FormItem = Form.Item
import OperationSearchForm, { OperationSearchFormProps } from 'components/OperationSearchForm'
import Section from 'components/Section/Section'

describe('Компонент PendingOperationsForm', () => {

  const props: OperationSearchFormProps = {}

  const shallowWrapper = shallow(<OperationSearchForm {...props} />)
  const renderWrapper = render(<OperationSearchForm {...props} />)

  it('отображает компоненты <FormItem/>', () => {
    expect(shallowWrapper.find(FormItem)).toHaveLength(5)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
