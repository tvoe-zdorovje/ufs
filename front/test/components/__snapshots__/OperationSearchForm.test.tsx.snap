// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент PendingOperationsForm совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row"
  >
    <span
      class="label"
    >
      Введите данные по клиенту
    </span>
  </div>
  <div
    class="ant-row"
    style="margin-top:15px;"
  >
    <div
      class="ant-col-4"
    >
      <span
        class="label"
      >
        Фамилия
      </span>
    </div>
    <div
      class="ant-col-16 ant-col-offset-1"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control "
          >
            <input
              class="ant-input ant-input-lg"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-4"
    >
      <span
        class="label"
      >
        Имя
      </span>
    </div>
    <div
      class="ant-col-16 ant-col-offset-1"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control "
          >
            <input
              class="ant-input ant-input-lg"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-4"
    >
      <span
        class="label"
      >
        Отчество
      </span>
    </div>
    <div
      class="ant-col-16 ant-col-offset-1"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control "
          >
            <input
              class="ant-input ant-input-lg"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin-top:15px;"
  >
    <span
      class="label"
    >
      Введите данные ДУЛ: 
    </span>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-4"
    >
      <span
        class="label"
      >
        Тип ДУЛ
      </span>
    </div>
    <div
      class="ant-col-16 ant-col-offset-1"
    >
      <div
        class="ant-select ant-select-enabled"
        style="display:block;"
      >
        <div
          aria-autocomplete="list"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-select-selection
                    ant-select-selection--single"
          role="combobox"
          tabindex="0"
        >
          <div
            class="ant-select-selection__rendered"
          >
            <div
              class="ant-select-selection-selected-value"
              style="display:block;opacity:1;"
              title="Паспорт гражданина РФ"
            >
              Паспорт гражданина РФ
            </div>
          </div>
          <span
            class="ant-select-arrow"
            style="user-select:none;-webkit-user-select:none;"
            unselectable="unselectable"
          >
            <b />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-4 label"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control "
          >
            <input
              class="ant-input ant-input-lg"
              placeholder="Серия"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col-16 ant-col-offset-1"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control "
          >
            <input
              class="ant-input ant-input-lg"
              placeholder="Номер"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
